python/spacewalk/satellite_tools/appstreams.py

```diff
@@ -286,7 +286,7 @@ def _parse_rpm_name(nevra):
         pattern = re.compile(
             r"(?P<name>[a-zA-Z0-9._+-]+)-"
             r"(?P<epoch>\d+:)?"
-            r"(?P<version>[a-zA-Z0-9._-]+)-"
+            r"(?P<version>[a-zA-Z0-9._-~]+)-"
```

Comment:
Is a hyphen/dash inside the version string valid? openSUSE's packaging documentation says it's not available, does that differ in the RH world?
>RPM uses the dash to separate the package name, the package version and the distro-level re-issue ("release") counter. (`rpm -qa` for example emits coreutils-9.0-1.x86_64.) The dash itself is unavailable for use within the version or release substring, as are a few other characters like the asterisk. [...]

https://en.opensuse.org/openSUSE:Package_versioning_guidelines#Unusual_characters

Reply:
The real problem is that there's no actual standard between distros and like also mentioned there in the document, some distros are pretty liberal with this.

For "hyphens in version" case, Ubuntu/Debian is the offender. Here are some examples:
```
ipxe-qemu-256k-compat-efi-roms-1.0.0+git-20150424.a25a16d-0ubuntu4.all-deb.deb
ipxe-1.0.0+git-20190109.133f4c4-0ubuntu3.all-deb.deb
ipxe-qemu-1.0.0+git-20190109.133f4c4-0ubuntu3.all-deb.deb
grub-ipxe-1.0.0+git-20190109.133f4c4-0ubuntu3.all-deb.deb
lib32gcc1-10-20200411-0ubuntu1.amd64-deb.deb
libx32gcc1-10-20200411-0ubuntu1.amd64-deb.deb
```

I should probably add some more test cases reflecting this.

Reply:
Is appstreams.py also used by Debian repos? I thought this was RPM-only.

Package version schemas definitely differ between packaging formats and we can't have one way to handle all of them. The same applies to version comparisons, where has their own rules.

Reply:
`libsolv` has some version comparison algorithms from which  maybe some reasonable expectations for a version string can be extracted:

https://github.com/openSUSE/libsolv/blob/master/src/evr.c

Reply:
I can have a look at it if it covers all the necessary formats. But in any case, further changes are not in scope for this fix and I don't want to risk any regressions without testing extensively. So I'm just going to merge this one as is.